app/battle.py

```diff
@@ -0,0 +1,24 @@
+from app.knights import formation_knight, Knight
+
+
+def battle_pre(knight1: Knight, knight2: Knight) -> dict:
+    knight1.hp -= knight2.power - knight1.protection
+    knight2.hp -= knight1.power - knight2.protection
+    if knight1.hp <= 0:

```

Comment:
do it in one line

Reply:
I didn’t quite understand that in one line